### PR TITLE
search: add 'editor' type in the search result

### PR DESCRIPTION
**Pull request by prabalsingh24**
### Problem
Added `Editor` in the search result

### Solution
Screenshot_2020-05-08 Search Results – BookBrainz

### CODE REVIEW

**Comment by prabalsingh24** [comment]
TODO: 
1. I haven't changed `collection` to `type` in the routes yet.
2. If you update your username, it is not getting updated in the navbar. (it is getting updated in the search result though)

**Comment by prabalsingh24** [comment]
You'll have to do a re-indexing first.

**Comment by MonkeyDo** [comment]
It does look like after changing the username it's reindexed properly and I find it in the search results.

Do you want to change collection to type in this PR, or in a new one?

**Comment by prabalsingh24** [comment]
> It does look like after changing the username it's reindexed properly and I find it in the search results.

But it's not getting updated in the NavBar. Right top corner.

**Comment by prabalsingh24** [comment]
sessions are stored in redis right?

**Comment by prabalsingh24** [comment]
I'll make the change in this PR only.

**Comment by MonkeyDo** [comment]
> > It does look like after changing the username it's reindexed properly and I find it in the search results.
> 
> But it's not getting updated in the NavBar. Right top corner.

Huh, that's right. Sorry I read too fast :p
I just checked on test.bb and it looks like the issue is not linked to this PR at all, so not a blocker. I'll create a ticket.

Sessions are indeed stored on redis. Do you think they might need to be invalidated or refreshed for the username to appear correctly?

**Comment by prabalsingh24** [comment]
I'll have a look. I will change `All Entity` --> `All Entities` and `collection` --> `type`

File: src/server/helpers/search.js [hunk 1]
```diff
@@ -358,16 +371,36 @@ export async function generateIndex(orm) {
...
+	/** To index names, we use aliasSet.aliases.name and bbid, which Editors don't have.
+	 * We massage the editor to return a similar format as BB entities
+	 */
+	const processedEditors = editors.map((editor) => new Object({
+		aliasSet: {
+			aliases: [
+				{name: editor.name}
+			]
+		},
+		bbid: editor.id,
```
Comment (prabalsingh24) [inline src/server/helpers/search.js:389]: Do you approve of this?
By adding `bbid` here,  I avoided adding a special case for `editor` here. [search-table-row-link](https://github.com/bookbrainz/bookbrainz-site/blob/a8aa31b49e8cb9e63f9ddd71f9bf37fef528e350/src/client/components/pages/parts/search-results.js#L77)
The same can be done for collections too..

**Comment by prabalsingh24** [comment]
I changed it. I think this can be merged now :)

**Comment by MonkeyDo** [comment]
I don't see the changes to `src/client/entity-editor/name-section/actions.js`. Maybe you forgot to push?

**Comment by prabalsingh24** [comment]
> I don't see the changes to `src/client/entity-editor/name-section/actions.js`. Maybe you forgot to push?

Oops. Yeah I forgot to push :P

**Comment by prabalsingh24** [comment]
I'll have to change some tests too :)

**Pull request merged**